### PR TITLE
Support `Statement.add()` for multiple bindings

### PR DESCRIPTION
**Pull request by mirromutth**
See issue https://github.com/jasync-sql/jasync-sql/issues/110

### CODE REVIEW

File: r2dbc-mysql/src/main/java/ExtendedStatement.kt [hunk 1]
```diff
@@ -5,15 +5,17 @@ import io.r2dbc.spi.Result
 import io.r2dbc.spi.Statement
 import org.reactivestreams.Publisher
 import reactor.core.publisher.Mono
+import reactor.core.publisher.toFlux
 import reactor.core.publisher.toMono
-import java.util.concurrent.CompletableFuture
 import java.util.function.Supplier
 import com.github.jasync.sql.db.Connection as JasyncConnection
 
-class SimpleStatement(private val clientSupplier: Supplier<JasyncConnection>, private val sql: String) : Statement {
```
Comment (oshai) [inline r2dbc-mysql/src/main/java/ExtendedStatement.kt]: I think calling it StatementImpl might be more appropriate (or JasyncStatement)

File: r2dbc-mysql/src/main/java/Bindings.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package com.github.jasync.r2dbc.mysql
+
+/**
+ * Statement parameters bindings.
+ */
+internal class Bindings {
+
+    private val bindings: MutableList<Map<Int, Any?>> = mutableListOf()
```
Comment (oshai) [inline r2dbc-mysql/src/main/java/Bindings.kt]: how about calling it paramBindings?

File: r2dbc-mysql/src/main/java/ExtendedStatement.kt [hunk 3]
```diff
@@ -57,40 +63,35 @@ class SimpleStatement(private val clientSupplier: Supplier<JasyncConnection>, pr
...
-            if (selectLastInsertId) {
-                queried.flatMap { result ->
-                    connection.sendQuery("SELECT LAST_INSERT_ID() AS $generatedKeyName")
-                        .toMono()
-                }
+        return Mono.fromSupplier(clientSupplier).flatMapMany { connection ->
+            if (isPrepared) {
+                val allParams = bindings.all().asSequence().mapIndexed { i, binding ->
+                    (0 until binding.size).map {
+                        if (it in binding) {
```
Comment (oshai) [inline r2dbc-mysql/src/main/java/ExtendedStatement.kt]: nice

**Comment by oshai** [comment]
@mp911de - can you please review that as well?

File: r2dbc-mysql/src/main/java/Bindings.kt (hunk 2 above)
Comment (mp911de) [reply r2dbc-mysql/src/main/java/Bindings.kt]: `bindings` lines nicely up with what we have in the other drivers as well.

File: r2dbc-mysql/src/main/java/ExtendedStatement.kt (hunk 1 above)
Comment (mp911de) [reply r2dbc-mysql/src/main/java/ExtendedStatement.kt]: 👍 for `JasyncStatement`

**Pull request merged**